src/lib/es5.d.ts

```diff
@@ -180,6 +180,18 @@ interface ObjectConstructor {
       * Prevents the modification of existing property attributes and values, and prevents the addition of new properties.
       * @param o Object on which to lock the attributes.
       */
+    freeze<T>(a: T[]): ReadonlyArray<T>;
+
+    /**
+      * Prevents the modification of existing property attributes and values, and prevents the addition of new properties.
+      * @param o Object on which to lock the attributes.
+      */
+    freeze<T>(f: (...args: any[]) => T): (...args: any[]) => T;

```

Comment:
@mhegazy Can you use `freeze<T extends Function>(f: T): T;` instead of the following your definition?

```diff
 +    freeze<T extends (...args: any[]) => any>(f: T): T;
 +
 +    /**
 +      * Prevents the modification of existing property attributes and values, and prevents the addition of new properties.
 +      * @param o Object on which to lock the attributes.
 +      */
 +    freeze<T extends new (...args: any[]) => any>(c: T): T;
 +
 +    /**
 +      * Prevents the modification of existing property attributes and values, and prevents the addition of new properties.
 +      * @param o Object on which to lock the attributes.
 +      */
 +    freeze<T>(f: (...args: any[]) => T): (...args: any[]) => T;
```